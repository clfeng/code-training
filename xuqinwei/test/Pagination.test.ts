import { shallowMount, mount } from "@vue/test-utils";
import { expect, test, vi } from "vitest";
import Pagination from "../src/components/simplePagination/index";


test("mount component", () => {
  const handleInputChange = vi.fn()
  const wrapper = shallowMount(Pagination, {
    props: {
      current: 1,
      pageSize: 10,
      total: 47,
    },
    // setup(props, { attrs, emit, slots }) {

    // return { handleInputChange }
    // }
  });

  expect(wrapper.html()).toMatchSnapshot();
});