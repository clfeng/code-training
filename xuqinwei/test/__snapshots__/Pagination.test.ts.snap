// Vitest Snapshot v1

exports[`mount component 1`] = `
"<ul class=\\"si-pagination\\">
  <li class=\\"si-pagination__item prve-page\\">&lt;</li>
  <li class=\\"si-pagination__item\\">1</li>
  <li class=\\"si-pagination__item\\">2</li>
  <li class=\\"si-pagination__item\\">3</li>
  <li class=\\"si-pagination__item\\">4</li>
  <li class=\\"si-pagination__item\\">5</li>
  <li class=\\"si-pagination__item next-page\\">&gt;</li>
  <div><span>跳至</span><input class=\\"si-pagination__input\\"><span>页</span></div>
</ul>"
`;
